pkg/transport/generic_transport_controller.go

```diff
@@ -618,3 +913,24 @@ func cleanObject(object *unstructured.Unstructured) *unstructured.Unstructured {
 
 	return objectCopy
 }
+
+func MapCopyIntoFiltered[Key comparable, Val any](dest, src map[Key]Val, pass func(Key) bool) {
+	for key, val := range src {
+		if pass(key) {
+			dest[key] = val
+		}
+	}
+}
+
+func MapEqual[Key, Val comparable](map1, map2 map[Key]Val) bool {
+	if len(map1) != len(map2) {
+		return false
+	}
+	for key, val1 := range map1 {
+		val2, have := map2[key]
+		if val1 != val2 || !have {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
seems like these functions are not used anywhere.. can we remove it?

Reply:
Yes, done.